### ISSUE TITLE
File encrypt and decrypt methods must have different input and outputFilename

### ISSUE THREAD

**Issue by supertanuki** [defuse/php-encryption#403]
Hello,

Great library, thanks.

I spent several hours figuring out why my code is not working. I was passing same path for input and output file to file encrypt or decrypt methods.
I think it would be useful to have an exception in this case:

```php
    private static function encryptFileInternal($inputFilename, $outputFilename, KeyOrPassword $secret)
    {
        if ($inputFilename === $outputFilename) {
            throw new Ex\IOException('Input and outputFilename must be different.');
        }
```

```php
    private static function decryptFileInternal($inputFilename, $outputFilename, KeyOrPassword $secret)
    {
        if ($inputFilename === $outputFilename) {
            throw new Ex\IOException('Input and outputFilename must be different.');
        }
```

If you are ok, I will open the PR.

**Comment by supertanuki** [comment]
There is a way to check that `/tmp/foo.txt` and `../../../../../../../tmp/foo.txt` are same path? is it overkill ?

**Comment by erikbaan** [comment]
use PHP `realpath`  http://www.php.net/realpath

**Comment by defuse** [comment]
Fixed in #480
